app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/provision_workflow/general.rb

```diff
@@ -6,7 +6,12 @@ module ManageIQ::Providers::IbmCloud::VPC::CloudManager::ProvisionWorkflow::Gene
   # @param _options [void]
   # @return [Hash] Hash with ems_ref as key and name as value.
   def provision_type_to_profile(_options = {})
-    @provision_type_to_profile ||= index_dropdown(ar_ems.flavors)
+    return {} if ar_ems.nil?
+
+    @provision_type_to_profile ||= begin
+      flavors = resources_for_ui[:ems] ? ar_ems.flavors : ManageIQ::Providers::IbmCloud::VPC::CloudManager::Flavor.all
```

Comment:
Is it possible to ever make it to `ManageIQ::Providers::IbmCloud::VPC::CloudManager::Flavor.all` if we're returning if `ar_ems.nil?` above?

Reply:
Doesn't seem like it's possible. Should I remove the ternary operator and similar occurrences in other methods?